core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -160,7 +160,11 @@ class GroupMetadataManager(brokerId: Int,
 
   // return true iff group is owned and the group doesn't exist
   def groupNotExists(groupId: String) = inLock(partitionLock) {
-    isGroupLocal(groupId) && (!groupMetadataCache.contains(groupId) || groupMetadataCache.get(groupId).is(Dead))
+    isGroupLocal(groupId) && (!groupMetadataCache.contains(groupId) || {
```

Comment:
Can you write a short test case to make sure this function works correctly. Also, I think this is a bit more concise:
```scala
    isGroupLocal(groupId) && getGroup(groupId).forall { group =>
      group.inLock(group.is(Dead))
    }
```

Reply:
Thanks for the code improvement suggestion. I added a basic unit test in the new commit.